outlines/text/generate/sequence.py

```diff
@@ -247,8 +244,8 @@ def vectorized_random_choice(
 
     """
 
-    cumsum = np.expand_dims(p.cumsum(axis=-1), 0)
-    rand = rng.random((samples,) + p.shape[:-1])
-    idx = vsearchsorted(cumsum, rand)
+    cumsum = torch.unsqueeze(p.cumsum(axis=-1), 0)
+    rand = torch.rand((samples,) + p.shape[:-1] + (1,), generator=rng)
+    idx = (cumsum < rand).sum(axis=-1)
```

Comment:
random generation fails when cumsum and rand are on different devices. 
I don't know how to get this to work. 

This works on CPU
```
rng = torch.Generator()
rand = torch.rand((1, 1), generator=rng)
```

This does not work:
```
rng = torch.Generator(device='cuda')
rand = torch.rand((1, 1), generator=rng)
```

neither does this:
```
rng = torch.Generator()
rand = torch.rand((1, 1), generator=rng, device='cuda')
```

this works, but then we cannot use a rng generator 
```
rand = torch.rand((1, 1), device='cuda')
```